pkg/engine/lifecycle_test.go

```diff
@@ -1554,8 +1556,37 @@ func TestRefreshWithDelete(t *testing.T) {
 	}
 }
 
+func pickURN(urns []resource.URN, target string) resource.URN {
+	switch target {
+	case "resA":
+		return urns[0]
+	case "resB":
+		return urns[1]
+	case "resC":
+		return urns[2]
+	default:
+		panic("Invalid target: " + target)
+	}
+}
+
 // Tests that dependencies are correctly rewritten when refresh removes deleted resources.
 func TestRefreshDeleteDependencies(t *testing.T) {
+	names := []string{"resA", "resB", "resC"}
+
+	// Try refreshing a stack with every combination of the three above resources as a target to
+	// refresh.
+	subsets := combinations.All(names)
+
+	// combinations.All doesn't return the empty set.  So explicitly test that case (i.e. test no
+	// targets specified)
+	validateRefreshDeleteCombination(t, names, []string{})
+
+	for _, subset := range subsets {
+		validateRefreshDeleteCombination(t, names, subset)
+	}
```

Comment:
@pgavlin lmk if this is "too cute" in your testing lib.  I wanted to be able to bulk test someone trying every combinatoin of delete targets.  i.e. "A", "B", "A,B", "C", "A,C", "B,C", "A,B,C".  Was a lot easier to just write this way than duplicating tests.

Reply:
Nope, this is great